lib/travis/build/git/clone.rb

```diff
@@ -98,6 +98,7 @@ def fetch_ref?
           end
 
           def checkout
+            return fetch_head_alternative if data.repository[:vcs_type].to_s != '' && data.repository[:vcs_type].to_s != 'GithubRepository' && data.pull_request
```

Comment:
data.repository[:vcs_type].to_s != '' && data.repository[:vcs_type].to_s != 'GithubRepository' && data.pull_request
I propose to create check method

def vcs_pull_request?
  data.repository[:vcs_type].present? &&  data.repository[:vcs_type] != 'GithubRepository' && data.pull_request
end

Reply:
brilliant idea!